share/spice/twitch/channel_info/twitch_channel_info.js

```diff
@@ -0,0 +1,107 @@
+(function (env) {
+    "use strict";
+    
+    var ID = 'twitch_channel_info';
+    var STREAM_ENDPOINT = '/js/spice/twitch/stream/';
+    var BASE = 'https://www.twitch.tv/';
+    
+    env.ddg_spice_twitch_channel_info = function(api_result) {
+        // Validate the response
+        if (!api_result || api_result.error) {
+            return Spice.failed('twitch_channel_info');
+        }
+        
+        var name = api_result.name.trim().toLowerCase();
+        var created_at = new Date(api_result.created_at);
+        
+        
+        // Render the response
+        Spice.add({
+            id: ID,
+            name: "Streaming",
+            data: api_result,
+            meta: {
+                sourceName: "Twitch.tv",
+                sourceUrl: BASE + name,
+                sourceIconUrl: BASE + 'favicon.ico'
+            },
+            normalize: function(item) {
+                var getLink = function(url, label){
+                    return '<a href="' + url + '" title="">' + label + '</a>';
+                }
+                return {
+                    url: BASE + item.name,
+                    title: item.name,
+                    subtitle: [
+                        'Offline',
+                        new Handlebars.SafeString('Playing <strong>' + getLink(BASE + 'directory/game/' + item.game, item.game) + '</strong>')
+                    ],
+                    game: item.game,
+                    game_url: BASE + item.game, 
+                    image: item.video_banner,
+                    description: item.status,
+                    infoboxData: [
+                        {
+                            heading: 'Channel info'
+                        },
+                        {
+                            label: 'Created at',
+                            value: created_at.toISOString().slice(0, 10)
+                        },
+                        {
+                            label: 'Followers',
+                            value: item.followers.toLocaleString()
+                        },
+                        {
+                            label: 'Views',
+                            value: item.views.toLocaleString()
+                        }
+                    ]
+                };
+            },
+            templates: {
+                group: 'info',
+                options: {
+                    content: Spice.twitch_channel_info.content
+                }
+            },
+            // Check stream status after render and load preview if it's LIVE
+            onItemShown: function(item) {
```

Comment:
Let's simplify this:

```js
onItemShown: function(item) {

    $.getJSON(STREAM_ENDPOINT + name, function(data) {

        var obj = {
            subtitle: [
                "Offline",
                playing
            ]
        };

        if (Object.keys(data).length && data.stream) {

            obj.subtitle[0] = "Online";

            // Update viewers count
            if (data.stream.viewers) {
                obj.subtitle.push( "Viewers: " + DDG.commifyNumber(data.stream.viewers) );
            }

            // Update stream preview
            if (data.stream.preview) {
                console.log(data.stream.preview);
                obj.image = data.stream.preview.medium;
            }
        }
        item.set(obj);
    });
}
```

This way we will update the model accordingly and let the template update itself. I've removed `subtitle` above so that it only get's set once we know what the status is instead of initially setting it to "Offline" then quickly changing it to "Online" when the 2nd API call returns.

Please also set `meta.rerender`:

```js
meta: {
    sourceName: "Twitch.tv",
    sourceUrl: BASE + name,
    sourceIconUrl: BASE + 'favicon.ico',
    rerender: ["image", "subtitle"]
},
```